File: src/impl/dataflow_pattern.cc
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

/*!
 * \file src/impl/dataflow_pattern.cc
 * \brief Dataflow pattern of RAF IR
 */

#include <tvm/node/repr_printer.h>

#include "raf/registry.h"
#include "raf/dataflow_pattern.h"

namespace raf {
namespace ir {

using tvm::ReprPrinter;

// Pattern sugars
DFPattern IsRelayConstant() {
  return RelayConstantPattern(make_object<RelayConstantPatternNode>());
}

DFPattern IsConstant(ObjectRef value) {
  ObjectPtr<ConstantPatternNode> n = make_object<ConstantPatternNode>();
  n->value = std::move(value);
  return RelayConstantPattern(n);
}

RAF_REGISTER_GLOBAL("raf.ir.dataflow_pattern.is_constant").set_body_typed(IsConstant);

}  // namespace ir
}  // namespace raf

File: python/raf/ir/dataflow_pattern.py
# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0

# pylint: disable=unused-import
"""The RAF Pattern Language and tooling."""
from raf._lib import (
    DFPattern,
    is_var,
    is_expr,
    is_op,
    is_tuple,
    is_tuple_get_item,
    is_if,
    is_let,
    wildcard,
    has_type,
    has_dtype,
    has_shape,
    has_attr,
    dominates,
)
from raf._ffi.ir.dataflow_pattern import is_constant
from raf._ffi.pass_ import dataflow_pattern_match as match
from raf._ffi.pass_ import dataflow_pattern_rewrite as rewrite
from raf._ffi.pass_ import dataflow_pattern_partition as partition
